### PR TITLE
feat(partition): add tooltip

### PR DESCRIPTION
**Pull request by monfera**
## Summary

Adds shape picking calculation, basic event handler and cursor location transform.
It may return multiple items, for example:

- multilayer treemap: it returns all hovered layers (can be less than the total number of layers if the user hovers over what's a gap at a granular level, but inside a rectangle at a less granular level)
- pie/sunburst: we may extend the logic such that it covers the entire pixel (now it covers the corner of the pixel, so currently it'll yield up to one shape)

shape pick

### Checklist

- [ ] ~~Any consumer-facing exports were added to `src/index.ts` (and stories only import from `../src` except for test data & storybook)~~
- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- [ ] ~~Proper documentation or storybook story was added for features that require explanation or tutorials~~
- [ ] Unit tests were updated or added to match the most common scenarios
- [x] Each commit follows the [convention](https://github.com/elastic/elastic-charts/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by robgil** [comment]
# [Codecov](https://codecov.io/gh/elastic/elastic-charts/pull/544?src=pr&el=h1) Report
> Merging [#544](https://codecov.io/gh/elastic/elastic-charts/pull/544?src=pr&el=desc) into [master](https://codecov.io/gh/elastic/elastic-charts/commit/ce4e84fac8b51861cde377303ecaf9038611158b&el=desc) will **decrease** coverage by `0.24%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/elastic/elastic-charts/pull/544?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master     #544      +/-   ##
==========================================
- Coverage   75.57%   75.33%   -0.25%     
==========================================
  Files         181      181              
  Lines        5691     5716      +25     
  Branches     1104     1109       +5     
==========================================
+ Hits         4301     4306       +5     
- Misses       1374     1394      +20     
  Partials       16       16              
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/elastic/elastic-charts/pull/544?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/elastic/elastic-charts/pull/544?src=pr&el=footer). Last update [1a73e4a...dfbac01](https://codecov.io/gh/elastic/elastic-charts/pull/544?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by monfera** [comment]
Added data collection for which individual data array elements make up the sunburst slice or treemap rectangle on @markov00's request. The collection and identification of the original `Datum` elements in the passed-on `data` array is done via their zero-based indices in the array. Here's an example in case of a three-layer sunburst:

indices

A new story is added, which has more than one input `Datum` for each (leaf) slice.

File: src/chart_types/partition_chart/state/selectors/tooltip.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+const getValueFormatter = (state: GlobalChartState) => {
+  const pieSpecs = getSpecsFromStore<PartitionSpec>(state.specs, ChartTypes.Partition, SpecTypes.Series);
+  return pieSpecs[0].valueFormatter;
+};
+
+const getLabelFormatters = (state: GlobalChartState) => {
+  const pieSpecs = getSpecsFromStore<PartitionSpec>(state.specs, ChartTypes.Partition, SpecTypes.Series);
+  return pieSpecs[0].layers;
+};
```
Comment (nickofthyme) [inline src/chart_types/partition_chart/state/selectors/tooltip.ts]: Seems like there is a cleaner way to do this but looks fine. Maybe just a `getFirstSpec` selector.

Comment (markov00) [reply src/chart_types/partition_chart/state/selectors/tooltip.ts]: fixed in 918939e

**Pull request merged**